### ISSUE TITLE
Failed Extraction with Library Loaded

### ISSUE THREAD

**Issue by Cykapath** [cadon/ARKStatsExtractor#906]
I have noticed this issue a few times before, I am unable to extract a dino but if I click new spreadsheet and extract the exact same export file, it works just fine. 

Example 1 (paracer extraction failed with spreadsheet loaded):
![example1](https://user-images.githubusercontent.com/49641683/56126876-db803100-5f73-11e9-9363-05db55c94060.png)

Example 2: (exact same export file, in a new blank spreadsheet file):
![example2](https://user-images.githubusercontent.com/49641683/56126877-de7b2180-5f73-11e9-8422-54d5e786ff8d.png)

**Comment by DodoCooker** [comment]
Likely a difference in your settings. Best  provide screenshots of the settings from "File/Settings" of your saved library and from the new empty one (you might even be able to see differences yourself already there to fix it).

**Comment by Cykapath** [comment]
Seems unlikely as I had been using this particular spreadsheet for 2 weeks without changing any settings and it only suddenly stopped working. Other times I have been using a spreadsheet for over a month or two and then the error suddenly occurs on some/most export files.

**Comment by cadon** [comment]
Can you upload the according export-file along with a screenshoft of the settings? It might be a rounding issue. Does this error occur always for certain files, or do files sometimes work and then don't?

**Comment by Cykapath** [comment]
Unfortunately I cleared my export folder. I play on small tribe servers. The error appears to occur once my spreadsheet gets to a certain size it appears, however this could just be a coincidence. In the most recent example, I had just imported some very high level (295 wild) direwolves that been tamed in the ice cave on extinction and after that, every import was broken, including dinos I already had imported a few hours ago. The wolves imported just fine but their names were in red text inside the library tab, which I couldn't resolve in any way. I hope this helps. Here are the settings I use:

![settings](https://user-images.githubusercontent.com/49641683/56531171-787e3380-654b-11e9-85b4-b0eece7b06f0.png)

**Comment by Cykapath** [comment]
I have just encountered this error once again, this time with a giga which I previously had no issues with:

![giga1](https://user-images.githubusercontent.com/49641683/56617421-bd6f9c00-6617-11e9-9e94-c3bb079a139b.png)

![giga2](https://user-images.githubusercontent.com/49641683/56617426-bf395f80-6617-11e9-81d2-9a120d43758b.png)

Here are my settings as well:

![settings](https://user-images.githubusercontent.com/49641683/56617460-d710e380-6617-11e9-876a-2de703eca388.png)

**Comment by VolatilePulse** [comment]
I believe the issue you are experience ING is one that has been floating around for a while and is intermittent. I have yet to find the cause for it yet but a temporary fix appears to be to restart ASB and try to extract the creature immediately. Let me know if this works. 

If you are interested, the issue appears to be a species index lookup. Essentially the program is trying to extract a species other than the Giga but shows it is try to extract the Giga. It's on the list to be fixed, but need to find the issue first. 

There also seems to be an existing issue when additional values are loaded for mods like classic flyers. This tends to throw the index off slightly and causes similar weird issues.

**Comment by Cykapath** [comment]
Ah, very interesting! I had spoken to a large number of other big breeders I know and they weren't aware of it, I must be unlucky! Re opening the program and extracting the creature immediately doesn't seem to work for me, however if I reset all my taming and breeding settings to defaults using the buttons in settings, that works. Please let me know if I can help in any other way to fix this bug.

**Comment by VolatilePulse** [comment]
I was in a bit of a rush at the moment when I saw repeated issues that lined up almost perfectly with what I saw previously. That doesn't appear to be your issue, but rather that you have BabyImprintingStatScaleMultiplier set to 2.0 instead of 1.0 to match your server settings. This value determines how much affect the imprinting has on certain stats. When you reset all of them, it extracts fine for that exact reason.

Let me know how that works for you.
